core/src/main/java/org/apache/james/core/MailAddress.java

```diff
@@ -665,4 +665,14 @@ private int parseDomain(StringBuffer dSB, String address, int pos) throws Addres
     public boolean isNullSender() {
         return false;
     }
+
+
+    /**
+     * Return <code>true</code> if the {@link MailAddress} equals the given address ignoring the case.
+     *
+     * @return nullsender
+     */
+    public boolean equalsIgnoreCase(String address) {
```

Comment:
We assume that two usernames can't be distinguished from case sensitivity ?
I can't find this rule in the RFC.

Reply:
In this RFC: https://tools.ietf.org/html/rfc5321#section-2.4
```
...
Mailbox domains follow normal DNS rules and are
   hence not case sensitive.
...
```

Reply:
I agree for domains, but for user/username? what is its status?

Reply:
the user part of a email address doesn't have rules. It's up to the authoritative server for this domain to decide how this part is handled.

Reply:
hence the "don't try to validate email address" legendary claim

Reply:
No, it doesn't work like that : you can't decide what are the comparison rules for the authoritative server of that domain, you can only decide for mail address you own.

Reply:
IMO we need to
 - Compare case sensitively the username
 - Compare in a case sensitive fashion the domain name

Case insensitiveness of domains is already e,beded as part of `Domain` object which MailAddress relies on.

Tests needs to be added in MailAddress tests:

```
a@d.com == a@D.com
A@d.com != a@d.com
```

But this commit should be dropped.

(BTW equals already do ignore case assertions...)

```
    @Override
    public final boolean equals(Object obj) {
        if (obj == null) {
            return false;
        } else if (obj instanceof String) {
            String theString = (String) obj;
            return toString().equalsIgnoreCase(theString);
        } else if (obj instanceof MailAddress) {
            MailAddress addr = (MailAddress) obj;
            if (isNullSender() && addr.isNullSender()) {
                return true;
            }
            return equalsIgnoreCase(getLocalPart(), addr.getLocalPart())
                && Objects.equals(getDomain(), addr.getDomain());
        }
        return false;
    }
```

So I wonder how `equalsIgnoreCase` differs from `equals`. We may need a (separate) ticket to fix this incorrect behaviour (James should support case variation in mailAdresss - or alike)

Maybe TDD would have avoided such duplicated code BTW...